### PR TITLE
Fix deployments loading from /nail/etc/services

### PR DESCRIPTION
**Pull request by kawaiwanyelp**
### Description
This PR primarily fixes deployments loading from `deployments.json`. The bug was that the function `deploymentsFromConfig` was loading the value of the "v2" key in `deployments.json` into a data structure that expected a "v2" key. I fixed the issue by loading the value into the proper data structure.

I also made some minor changes to the go-vet pre-commit hook since it doesn't actually work with multi-package repos.

### Testing
manual testing against actual `deployments.json` files in `/nail/etc/services`
make test, with a new against a test file for this specific case

### CODE REVIEW

**Comment by kawaiwanyelp** [comment]
@Baisang I get an error (i.e. "wtf") when I run your function, but it goes away with my fix. I'm not sure why it's different for you :P

**Comment by Baisang** [comment]
@kawaiwanyelp turns out I was just a million commits behind master. oops

File: pkg/deployments/deployments_test.go [hunk 1]
```diff
@@ -71,6 +74,55 @@ func TestMakeControlGroup(test *testing.T) {
...
+			},
+		},
+		"",
+		"  ",
+	)
+	if err != nil {
+		test.Error("Failed to serialize test deployment JSON")
+	}
+
+	tempDir, err := ioutil.TempDir(os.TempDir(), "paasta-tools-go-test-service-*")
```
Comment (Rob-Johnson) [inline pkg/deployments/deployments_test.go:103]: I think you need to clean up after this

Comment (kawaiwanyelp) [reply pkg/deployments/deployments_test.go:103]: I've added an `defer os.Remove` after this to clean up.

**Pull request merged**